api/types/access_request.go

```diff
@@ -642,6 +642,42 @@ func (u *AccessRequestUpdate) Check() error {
 	return nil
 }
 
+// RequestReasonMode can be either "required" or "optional". Empty-string is treated as "optional".
+// If a role has the request reason mode set to "required", then reason is required for all Access
+// Requests requesting roles or resources allowed by this role. It applies only to users who have
+// this role assigned.
+type RequestReasonMode string
+
+const (
+	// RequestReasonModeRequired indicates required mode. See [[RequestReasonMode]] godoc for
+	// more details.
+	RequestReasonModeRequired RequestReasonMode = "required"
+	// RequestReasonModeRequired indicates optional mode. See [[RequestReasonMode]] godoc for
+	// more details.
+	RequestReasonModeOptional RequestReasonMode = "optional"
+)
+
+// Required checks if this mode is "required". Empty mode is treated as "optional".
+func (m RequestReasonMode) Required() bool {
+	switch m {
+	case RequestReasonModeRequired:
+		return true
+	default:
+		return false
+	}
+}
+
+// Check validates this mode value. Note that an empty value is considered invalid.
+func (m RequestReasonMode) Check() error {
+	switch m {
+	case RequestReasonModeRequired, RequestReasonModeOptional:
+		return nil
+	default:
+		return trace.BadParameter("unrecognized request reason mode %q, must be one of: %q, %q",
+			m, RequestReasonModeRequired, RequestReasonModeOptional)
```

Comment:
Nit: Usually in cases like this I would make a list with valid reasons and use it here so we don't need to update this every time we add a new "mode" or need to include it in a message, etc.

```go
SupportedRequestReasonModes = []string{
    RequestReasonModeRequired,
    RequestReasonModeOptional,
}
```